submissions/chimoney-telegram-bot/src/start.js

```diff
@@ -0,0 +1,191 @@
+
+
+
+bot.start((ctx) => {
+
+    const checkUsers = JSON.parse(fs.readFileSync("users.json", "utf8"));
+    var searchUser = checkUsers.find(item => item.firstName === ctx.message.from.first_name);
+
+    if (!searchUser) {
+        var user_details = {}
+        user_details.firstName = ctx.message.from.first_name;
+        user_details.id = ctx.message.from.id;
+        user_details.username = ctx.message.from.username;
+
+        checkUsers.push(user_details);
+        fs.writeFileSync("users.json", JSON.stringify(checkUsers, null, 2));
+
+    }
+
+    if (ctx.message.text.includes("claim_")) {
+        var msg = ctx.message.text;
+
+        var paymentRef = msg.slice(13)
+
+        const readPayFile = JSON.parse(fs.readFileSync("payments.json", "utf8"));
+
+
+        var findPay = readPayFile.find(item => (item.paymentRef === paymentRef) && (item.receiverName === ctx.message.from.username));
+
+        if (findPay) {
+
+            var redeemLink = "https://dash.chimoney.io/redeem?chiRef=" + findPay.chiRef
+
+            var keyboard = Markup.inlineKeyboard([
+                Markup.button.url("Redeem", redeemLink)
+            ])
+
+            var msg = `Congrats!!!, You've received $${findPay.amount} from @${findPay.senderName}.\n\nRedeem Now:`;
+            ctx.reply(msg, keyboard)
+        }
+    }
+
+
+});
+
+
+bot.on('message', async (ctx) => {
+
+    if (ctx.message.chat.type === "group" || ctx.message.chat.type === "supergroup") {
+        if (ctx.message.text.toLowerCase().startsWith('/pay ')) {
+
+            const checkUsers = JSON.parse(fs.readFileSync("users.json", "utf8"));
+            var searchUser = checkUsers.find(item => item.firstName === ctx.message.from.first_name);
+
+            if (!searchUser) {
+                var keyboard = Markup.inlineKeyboard([
+                    Markup.button.url("Message Me", bot_url + "?start=firstMsg")
+                ])
+                ctx.reply("Please message me first before you can send chimoney to any one", keyboard)
+                return
+            }
+            var find_mentioned_user = ctx.message.entities.find(item => item.type === 'mention');
+
+            if (find_mentioned_user) {
+                var offset = find_mentioned_user.offset;
+                var length = find_mentioned_user.length;
+                var tagged_user = ctx.message.text.slice(6)
+
+                var mentioned_user = tagged_user.substring(0, length - 1)
+                var amount = tagged_user.slice(length);
+                var msg_from_id = ctx.message.from.id;
+                var msg_from_name = ctx.message.from.username;
+                var chatId = ctx.message.chat.id;
+
+
+                if (!isValidInteger(amount)) {
+                    ctx.reply("Enter a valid amount");
+                    return;
+                }
+
+                const config = {
+                    method: 'post',
+                    url: 'https://api-v2-sandbox.chimoney.io/v0.2/payment/initiate',
```

Comment:
Consider using a CHIMONEY_BASE_URL env variables for this so that this can be deployed in prod by changing the env variables